### PR TITLE
140042 PKS Wiremock Service

### PR DESCRIPTION
**Pull request by shirinbt14926**
- Added PKS post endpoint with wiremock in POS
- Added Build, publish and deployment steps for PKS wiremock project
- Added coverlet package into BESS unit test projects.

### CODE REVIEW

File: UKHO.PKSWireMockService.API/IWireMockService.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿namespace UKHO.PKSWireMock.API
+{
+    public interface IWireMockService
+    {
+        void Start();
+        void Stop();
+    }
+}
+
+
```
Comment (Nadeem14456) [inline UKHO.PKSWireMockService.API/IWireMockService.cs:10]:
```suggestion
namespace UKHO.PKSWireMock.API
{
    public interface IWireMockService
    {
        void Start();

        void Stop();
    }
}

```

Comment (Nadeem14456) [inline UKHO.PKSWireMockService.API/IWireMockService.cs:10]: inconsistent spacing 
```suggestion
﻿namespace UKHO.PKSWireMock.API
{
    public interface IWireMockService
    {
        void Start();

        void Stop();
    }
}

```

File: UKHO.PKSWireMockService.API/App.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿namespace UKHO.PKSWireMock.API
+{
+    public class App :IHostedService
```
Comment (Nadeem14456) [inline UKHO.PKSWireMockService.API/App.cs:3]:
```suggestion
    public class App : IHostedService
```

File: UKHO.PKSWireMockService.API/App.cs
Comment (Nadeem14456) [inline UKHO.PKSWireMockService.API/App.cs:1]: Avoid the use of underscore for private fields

File: UKHO.PKSWireMockService.API/WireMockService.cs
Comment (Nadeem14456) [inline UKHO.PKSWireMockService.API/WireMockService.cs:1]: Avoid the use of underscore for private fields

File: UKHO.PKSWireMockService.API/WireMockService.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            _server = WireMockServer.Start(_settings);
+
+            _server.Given(Request.Create().WithPath(PksUrl)
+                        .WithBody(string.Empty)
+                        .UsingPost())
+                    .RespondWith(Response.Create().WithStatusCode(415)
+                        .WithHeader(ContentType, ApplicationType)
+                        .WithBodyFromFile(Path.Combine(Path.GetFullPath(Path.Combine(Environment.CurrentDirectory)), "__files", "blankResponse.json")));
+          
+
```
Comment (Nadeem14456) [inline UKHO.PKSWireMockService.API/WireMockService.cs:69]: remove blank space

File: UKHO.PKSWireMockService.API/WireMockService.cs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            _logger.LogInformation($"WireMock.Net server settings {JsonConvert.SerializeObject(_settings)}");
+        }
+
+        private static string GetJsonData(string filePath)
+        {
+            using (var r = new StreamReader(filePath))
+            {
+                string jsonFile = r.ReadToEnd();
+                return jsonFile;
+            }
```
Comment (Nadeem14456) [inline UKHO.PKSWireMockService.API/WireMockService.cs:107]:
```suggestion
            using var fileStream = new StreamReader(filePath);
return  fileStream .ReadToEnd();
```

**Review by Nadeem14456 — CHANGES_REQUESTED** [review]

File: UKHO.PKSWireMockService.API/WireMockService.cs (hunk 3 above)
Comment (shirinbt14926) [reply UKHO.PKSWireMockService.API/WireMockService.cs:68]: Done

File: UKHO.PKSWireMockService.API/WireMockService.cs
Comment (shirinbt14926) [reply UKHO.PKSWireMockService.API/WireMockService.cs:1]: Done

File: UKHO.PKSWireMockService.API/App.cs
Comment (shirinbt14926) [reply UKHO.PKSWireMockService.API/App.cs:1]: Done

**Review by amogh14893 — APPROVED** [review]
Reviewed DevOps changes. LGTM

File: UKHO.PKSWireMockService.API/IWireMockService.cs (hunk 1 above)
Comment (shirinbt14926) [reply UKHO.PKSWireMockService.API/IWireMockService.cs:10]: Done

File: UKHO.PKSWireMockService.API/App.cs (hunk 2 above)
Comment (shirinbt14926) [reply UKHO.PKSWireMockService.API/App.cs:3]: Done

File: UKHO.PKSWireMockService.API/WireMockService.cs (hunk 4 above)
Comment (shirinbt14926) [reply UKHO.PKSWireMockService.API/WireMockService.cs]: Done

**Review by Nadeem14456 — APPROVED** [review]

**Review by DanOwenUKHO — APPROVED** [review]

**Pull request merged**